lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -65,6 +65,17 @@ def call(env)
           request_span.set_tag('http.method', request.request_method) if request_span.get_tag('http.method').nil?
           request_span.set_tag('http.url', request.path_info) if request_span.get_tag('http.url').nil?
           request_span.set_tag('http.status_code', status) if request_span.get_tag('http.status_code').nil? && status
+
+          # detect if the status code is a 5xx and flag the request span as an error
+          # unless it has been already set by the underlying framework
+          if status.to_s.start_with?('5') && request_span.status.zero?
+            request_span.status = 1
+            # in any case we don't touch the stacktrace if it has been set
+            if request_span.get_tag(Datadog::Ext::Errors::STACK).nil?
+              request_span.set_tag(Datadog::Ext::Errors::STACK, caller().join("\n"))
```

Comment:
I don't think `caller` is of much use here, this will only give the callstack up to this point which won't even include a single line of code that has been executed down the line in the middleware. I think that info is misleading, it's probably better to not have anything here.